### ISSUE TITLE
Denoise kernel size

### ISSUE THREAD

**Issue by mattcieslak** [nipreps/sdcflows#22]
I'm finding for some older phase1/phase2 fieldmaps that it's necessary to set the kernel size to 5 for this node: https://github.com/poldracklab/sdcflows/blob/0.0.2/sdcflows/workflows/phdiff.py#L98

Is there a downside to setting the default to 5? Or  should we make this a parameter?

**Comment by effigies** [comment]
Does it make sense to set a target smoothness instead of a straight kernel smoothing, as with [3dBlurToFWHM](https://afni.nimh.nih.gov/pub/dist/doc/program_help/3dBlurToFWHM.html)? Then we're less dependent on the quality of the input data.

**Comment by oesteban** [comment]
Hopefully, this will not be necessary after #14, but it doesn't hurt to first get this done.

**Comment by mattcieslak** [comment]
For now I have the kernel size only go up to 5 if there are two phase maps (#30) and the size stays 3 for phasediff images. #30 also re-organizes the workflows so that #14 can be used on these fieldmaps

**Comment by oesteban** [comment]
I've set 5 as the default for now, and I'm pretty happy with the results.

I'm also with @effigies that `3dBlurToFWHM` is a very cheap but effective solution to the problem.

Finally, we need to investigate the resampling when moving the fieldmap to the target EPI dataset because I'm under the impression that it is introducing some little spikes in it.

**Comment by mattcieslak** [comment]
Isn't `3dBlurToFWHM` doing Gaussian smoothing instead of a median filter? I thought the main purpose was to remove spikes, which would end up getting spread out spatially if you just smooth with a Gaussian kernel.

**Comment by oesteban** [comment]
Isolated spikes would be removed with a Gaussian filter easily - median filters are better for salt&pepper kind of noise. It doesn't hurt to try.

But yes, you are right - the median filter is just to remove spikes

**Comment by oesteban** [comment]
Superseded by #119.
